### DISCUSSION TITLE
subtract time?

### DISCUSSION THREAD

**Discussion by sconebonevideo** [ashmanix/obs-plugin-countdown#105] [Ideas]
is there a way to be able to subtract time, such as just putting a subtraction symbol the in requestpayload or would a requestType: sub_time need to be created?

**Comment by ashmanix** [comment]
So the short answer is that you couldn't subtract time from the timer...till today. I had to have another look at the websocket add_time and set_time features as they weren't working as I expected them to. In the process I've allowed time to be subtracted (well technically).

If you have a look at version 2.0.4 out today [here](https://github.com/ashmanix/obs-plugin-countdown/releases/tag/2.0.4) you should be able to download that version and give it a try though **be warned** that I've changed the way adding time and setting time works via websocket.

With the new setup you just send your usual `add_time` request but for the add time value put in minuses next to where you want to subtract time.
For example, to subtract 10 minutes and 10 seconds you would send a `time_to_add` value of `00:00:-10:-10` and that _should_ work. Man I hope it works. I'm beginning to hate the code for those functions 🤣

**Reply by sconebonevideo** [reply]
wow! thank you for looking into this! and it works perfectly! Also, the update resolved another issue; to add_time, we had to create a stop_all and play_all in order for the add_time to stick. (this was in Firebot and was not needed for Sammi)
